File: Phone.py
import json

def load_phonebook():
    try:
        with open('phonebook.json') as f:
            return json.load(f)
    except FileNotFoundError:
        return {}

def save_phonebook(phonebook):
    with open('phonebook.json', 'w') as f:
        json.dump(phonebook, f)

def add_contact(phonebook):
    name = input('Enter contact name: ')
    number = input('Enter contact number: ')
    phonebook[name] = number
    save_phonebook(phonebook)
    print(f'{name} has been added to the phone book.')

def search_contact(phonebook):
    name = input('Enter contact name to search for: ')
    if name in phonebook:
        print(f'{name}: {phonebook[name]}')
    else:
        print(f'{name} not found in the phone book.')

def edit_contact(phonebook):
    name = input('Enter contact name to edit: ')
    if name in phonebook:
        new_number = input(f'Enter new number for {name}: ')
        phonebook[name] = new_number
        save_phonebook(phonebook)
        print(f'{name}\'s number has been updated.')
    else:
        print(f'{name} not found in the phone book.')

def delete_contact(phonebook):
    name = input('Enter contact name to delete: ')
    if name in phonebook:
        del phonebook[name]
        save_phonebook(phonebook)
        print(f'{name} has been deleted from the phone book.')
    else:
        print(f'{name} not found in the phone book.')

def display_phonebook(phonebook):
    print('Phone Book:')
    for name, number in phonebook.items():
        print(f'{name}: {number}')

def main():
    phonebook = load_phonebook()

    while True:
        print('\nWelcome to the Phone Book App!')
        print('1. Add Contact')
        print('2. Search Contact')
        print('3. Edit Contact')
        print('4. Delete Contact')
        print('5. Display Phone Book')
        print('6. Exit')

        choice = input('Enter your choice: ')

        if choice == '1':
            add_contact(phonebook)
        elif choice == '2':
            search_contact(phonebook)
        elif choice == '3':
            edit_contact(phonebook)
        elif choice == '4':
            delete_contact(phonebook)
        elif choice == '5':
            display_phonebook(phonebook)
        elif choice == '6':
            break
        else:
            print('Invalid choice. Try again.')

if __name__ == '__main__':
    main()

File: phone.js
// Phonebook array to store contacts
let phonebook = [];

// Add contact function
function addContact() {
  let name = prompt("Enter name:");
  let phone = prompt("Enter phone number:");
  let contact = { name: name, phone: phone };
  phonebook.push(contact);
  playSound()
}

// Display phonebook function
function displayPhonebook() {
  let tableBody = document.getElementById("table-body");
  tableBody.innerHTML = "";
  phonebook.forEach(function(contact, index) {
    let row = `<tr>
                  <td>${index + 1}</td>
                  <td>${contact.name}</td>
                  <td>${contact.phone}</td>
                  <td class="actions">
                    <button onclick="editContact(${index})">Edit</button>
                    <button onclick="deleteContact(${index})">Delete</button>
                  </td>
                </tr>`;
    tableBody.innerHTML += row;
  });
}

// Search contact function
function searchContact() {
  let searchValue = document.getElementById("search-input").value.toLowerCase();
  let filteredContacts = phonebook.filter(function(contact) {
    return contact.name.toLowerCase().includes(searchValue) || contact.phone.includes(searchValue);
  });
  phonebook = filteredContacts;
  displayPhonebook();
}

// Edit contact function
function editContact(index) {
  let name = prompt("Enter new name:");
  let phone = prompt("Enter new phone number:");
  phonebook[index].name = name;
  phonebook[index].phone = phone;
  displayPhonebook();
}

// Delete contact function
function deleteContact(index) {
  phonebook.splice(index, 1);
  displayPhonebook();
}

// Event listeners
document.getElementById("add-btn").addEventListener("click", addContact);
document.getElementById("search-btn").addEventListener("click", searchContact);

// Play sound function
function playSound() {
  let audio = new Audio('drake_hotline_bling_lyrics_aac_45864.mp3');
  audio.play();
}

// Initial display
displayPhonebook();
